pulser/waveforms.py

```diff
@@ -794,6 +794,74 @@ def __init__(
             self._area / float(np.sum(self._norm_samples)) / 1e-3
         )
 
+    @classmethod
+    @parametrize
+    def from_max_val(
+        cls,
+        max_val: Union[float, Parametrized],
+        area: Union[float, Parametrized],
+        beta: Optional[Union[float, Parametrized]] = 14.0,
+    ) -> KaiserWaveform:
+        """Creates a Kaiser waveform with a threshold on the maximum value.
+
+        Instead of defining a duration, the waveform is defined by its area and
+        the maximum value. The duration is chosen so that the maximum value is
+        not surpassed, but approached as closely as possible.
+
+        Args:
+            max_val (float): The maximum value threshold (in rad/µs). If
+                negative, it is taken as the lower bound i.e. the minimum
+                value that can be reached. The sign of `max_val` must match the
+                sign of `area`.
+            area (float): The area under the waveform.
+            beta (Optional[float]): The beta parameter of the Kaiser window.
+                The default value is 14.
+        """
+        # Compute the ratio area / duration for a long duration
+        # and use this value for a first guess of the best duration
+
+        max_val = cast(float, max_val)
+        area = cast(float, area)
```

Comment:
To solve the case where `max_val` and `area` are negative, I think the way to go is to check that's the case here (raise an error if not), then store the sign in a variable and use the absolute values for `max_val` and `area`. Then when you create the final class that you return, you use the stored sign on the `area`.

Reply:
I agree.